File: solutions/go/darts/darts.go
package darts

import "math"

scores := map[int]int{
	10: 0,
	5: 1,
	1: 5
	-1: 10
}

func Score(x, y float64) int {
	distance := math.Sqrt(math.Pow(x, 2) + math.Pow(y, 2))
	if distance > 10 {
		return 0
	} else if distance > 5 {
		return 1
	} else if distance > 1 {
		return 5
	} else {
		return 10
	}
}

File: solutions/ruby/darts/darts.rb
class Darts
  include Math

  private

  attr_accessor :distance

  SCORES = {
    0..1 => 10,
    1..5 => 5,
    5..10 => 1,
    10.. => 0
  }.freeze

  def initialize(pos_x, pos_y)
    self.distance = Math.sqrt(pos_x * pos_x + pos_y * pos_y)
  end

  public

  def score
    SCORES.each { |range, points| return points if range.include?(distance) }
  end
end
